packages/flutter_tools/lib/src/compile.dart

```diff
@@ -889,3 +876,19 @@ class DefaultResidentCompiler implements ResidentCompiler {
     return _server.exitCode;
   }
 }
+
+/// Convert a file URI into a multiroot scheme URI if provided, otherwise
+/// return unmodified.
+@visibleForTesting
+String toMultiRootPath(Uri fileUri, String scheme, List<String> fileSystemRoots, bool windows) {
+  if (scheme == null || fileSystemRoots.isEmpty) {
+    return fileUri.toString();
+  }
+  final String filePath = fileUri.toFilePath(windows: windows);
+  for (final String fileSystemRoot in fileSystemRoots) {
+    if (filePath.startsWith(fileSystemRoot)) {
+      return scheme + filePath.substring(fileSystemRoot.length);
```

Comment:
This should be:

```
return scheme + '://' + filePath.substring(fileSystemRoot.length);
```

Reply:
do you have some examples of file schemes + paths used? this code is sort of fragile because it is assuming a particular string structure, I should really figure out how the Uri class should be used

Reply:
discussed offline